Ponca/src/SpatialPartitioning/KdTree/kdTree.h

```diff
@@ -245,10 +206,29 @@ class KdTreeBase
         m_min_cell_size = min_cell_size;
     }
 
-    // Internal ----------------------------------------------------------------
-protected:
-    inline void build_rec(NodeIndexType node_id, IndexType start, IndexType end, int level);
-    inline IndexType partition(IndexType start, IndexType end, int dim, Scalar value);
+    // Index mapping -----------------------------------------------------------
+public:
+    /// Return the point index associated with the specified sample index
+    inline IndexType point_from_sample(IndexType sample_index) const
```

Comment:
Most of the code of Ponca use the following case

```suggestion
    inline IndexType pointFromSample(IndexType sample_index) const
```